dataclass_factory/serializers.py

```diff
@@ -221,5 +221,15 @@ def create_serializer_impl(factory, schema: Schema, debug_path: bool,
     if is_collection(class_):
         item_serializer = get_lazy_serializer(factory)
         return get_collection_serializer(item_serializer)
-    else:
-        return get_vars_serializer(factory)
+
+    if isinstance(class_, type):
+        if instance_wont_have_dict(class_):
+            raise ValueError(f"Can not create serializer for {class_}")
+
+        if hasattr(class_, '__slots__') and '__dict__' in class_.__slots__:
```

Comment:
isn't it the same as `instance_wont_have_dict`?

Reply:
No. It could be a class without __slots__ and __dict__, such as built-in classes and Decimal in CPython.